File: frontend/demo/component/avatar/avatar-name.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/avatar';
import { html, LitElement } from 'lit';
import { customElement, state } from 'lit/decorators.js';
import { getPeople } from 'Frontend/demo/domain/DataService';
import { applyTheme } from 'Frontend/demo/theme';
import type Person from 'Frontend/generated/com/vaadin/demo/domain/Person';

@customElement('avatar-name')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  @state()
  private person: Person | undefined;

  protected override async firstUpdated() {
    const { people } = await getPeople({ count: 1 });
    this.person = people[0];
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-avatar .name="${`${this.person?.firstName} ${this.person?.lastName}`}">
      </vaadin-avatar>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/avatar/AvatarName.java
package com.vaadin.demo.component.avatar;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.demo.domain.DataService;
import com.vaadin.demo.domain.Person;
import com.vaadin.flow.component.avatar.Avatar;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;

@Route("avatar-name")
public class AvatarName extends Div {

  private Person person = DataService.getPeople(1).get(0);

  public AvatarName() {
    String name = person.getFirstName() + " " + person.getLastName();

    // tag::snippet[]
    Avatar avatarName = new Avatar(name);
    // end::snippet[]

    add(avatarName);
  }

  public static class Exporter extends DemoExporter<AvatarName> { // hidden-source-line
  } // hidden-source-line
}
